puppetexplore.py

```diff
@@ -74,6 +75,9 @@ def d42_update(dev42, nodes, options, static_opt, mapping, from_version='3', pup
 
         node_name = node['hostname']
 
+        if node_name == '' or node_name is None:
```

Comment:
this might not be needed since the updatedevice() method in device42.py now requires name of a device.